src/domain.rs

```diff
@@ -122,6 +136,7 @@ impl<T> WithMut<T> for core::sync::atomic::AtomicPtr<T> {
 /// structure be `'static`. If you wish to avoid that bound, you'll need to construct your own
 /// `Domain` for each instance of your data structure so that all the guarded data is reclaimed
 /// when your data structure is dropped.
+#[non_exhaustive]
```

Comment:
Implementers of Domain::Singleton relies on the fact that it is impossible to construct a Domain without a reference to some instance of F. This is an attempt to make explicit that assumption and avoid accidentally make other forms of construct Domain possible.

Even with this attribute, this guarantee is very fragile and at any time a new contributor can expose another way of building a Domain without noticing the side effect. It's very important to make this clear to new contributors, but I'm not sure how to document this restriction.

Reply:
I totally misunderstood how the visibility of the constructor works! Now I see that because family is a private field there is no way somebody use `Domain { .. }` outside this crate.